packages/app/src/systems/Chains/eth/hooks/useManageEthAssets.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+import { isAddress } from 'viem';
+import { useToken } from 'wagmi';
+
+import { useAssets } from './useAssets';
+
+export const useManageEthAssets = (newAssetAddress: string) => {
+  const { assets, handlers } = useAssets();
+
+  const { data, isError, isLoading } = useToken({
+    address: newAssetAddress as `0x${string}`,
+  });
+
+  const { filteredAssets, doesAssetExist } = useMemo(() => {
+    const isValidAddress = isAddress(newAssetAddress);
+    if (isValidAddress) {
+      const filteredAssets = assets.filter(
+        (asset) => asset.address === newAssetAddress
+      );
+      return { filteredAssets, doesAssetExist: filteredAssets.length };
+    }
+    const newAssetsArray = assets.filter((asset) =>
```

Comment:
avoid naming variables with type of it..`Array` `Obj` `String` etc...

try to name thinking on what's the purpose of the code, instead of what the code is doing

```suggestion
    const queriedAssets = assets.filter((asset) =>
```